protocols/request-response/src/lib.rs

```diff
@@ -677,7 +677,24 @@ where
         }
     }
 
-    fn on_dial_failure(&mut self, DialFailure { peer_id, .. }: DialFailure) {
+    // Removes the connection for a peer if it exists.
+    fn remove_connection_for_peer(&mut self, peer: PeerId, connection_id: ConnectionId) {
```

Comment:
This is unrelated to adding the field, right?

Reply:
Oops I thought I split the commits, must hav messed something up. I'll get this removed from this PR and update #4777  to not depend on these changes.